app/views/solutions/show.html.haml

```diff
@@ -64,15 +62,19 @@
           .react
             %h3 React to this solution
             =link_to [:my, :reactions, solution_id: @solution, emotion: 'like'], class: "like #{@user_reaction.try(:emotion) == "like" ? "selected" : ''}", method: :post, remote: true do
-              .hover Well done
-              =icon 'thumbs-up fa-fw', 'Thumbs up'
+              .hover Nice work!
+              =reaction_like_icon
             =link_to [:my, :reactions, solution_id: @solution, emotion: 'love'], class: "love #{@user_reaction.try(:emotion) == "love" ? "selected" : ''}", method: :post, remote: true do
               .hover Love it!
-              =icon 'heart fa-fw', 'Heart'
+              =reaction_love_icon
             =link_to [:my, :reactions, solution_id: @solution, emotion: 'genius'], class: "genius #{@user_reaction.try(:emotion) == "genius" ? "selected" : ''}", method: :post, remote: true do
               .hover Genius!
-              =graphical_image "genius.png", class: 'not-selected'
-              =graphical_image "genius-turquoise.png", class: 'selected'
+              =reaction_genius_icon
+
+        -if @solution.allow_comments? || @solution.comments.size > 0
```

Comment:
```suggestion
        -if @solution.allow_comments? || @solution.comments.any?
```

Reply:
Does any work with `size` under the hood, or does it automatically do `length`? (They do different things in Rails if you're not familiar).

Reply:
This is what I learned about it, but I can't read the source code to confirm:  it behaves like `size` in regard to that it doesn't recalculate if it already has the value. That's what you referred to, right? I also think it will immediately return false for an empty array, so not evaluating anything.

Reply:
So in Rails, `size` does `length` if the array is materialised, and `count` (a SQL `count`) if it's not.

So:
- `Solution.size == Solution.count`
- `Solution.to_a.size == Solution.length`

So the question is whether `any?` does `.length` or `.size` in the context of Rails.